atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -331,18 +327,26 @@ public String getNamespaceString() {
             return keyValueServiceNamespace;
         } else {
             Preconditions.checkState(keyValueService() instanceof InMemoryAtlasDbConfig,
-                "Expecting KeyvalueServiceConfig to be instance of InMemoryAtlasDbConfig, found %s",
+                    "Expecting KeyvalueServiceConfig to be instance of InMemoryAtlasDbConfig, found %s",
                     keyValueService().getClass());
             if (timelock().isPresent()) {
-                // Special case - empty timelock and empty namespace/keyspace does not make sense
-                Preconditions.checkState(timelock().get().client().isPresent(),
-                        "For InMemoryKVS, the TimeLock client should not be empty");
-                return timelock().get().client().get();
+                return timelock().get().client()
+                        .orElseThrow(() ->
+                                new IllegalStateException("For InMemoryKVS, the TimeLock client should not be empty"));
             }
             return UNSPECIFIED_NAMESPACE;
         }
     }
 
+    @Value.Derived
+    @JsonIgnore
+    public String getNamespaceString() {
```

Comment:
this pattern for initialising fields isn't thread safe, I think we get away with it here but we should be careful

Reply:
Yes, agreed that this is an unsafe check-then-act pattern, but the reinitialization of the namespace should not be worrisome here.